lib/dor/services/client/object.rb

```diff
@@ -1,25 +1,50 @@
 # frozen_string_literal: true
 
-require 'nokogiri'
-require 'deprecation'
+require 'dor/services/client/files'
+require 'dor/services/client/release_tags'
+require 'dor/services/client/sdr'
+require 'dor/services/client/workflow'
+require 'dor/services/client/workspace'
 
 module Dor
   module Services
     class Client
       # API calls that are about a repository object
       class Object < VersionedService
-        # @param object [String] the pid for the object
-        def initialize(connection:, version:, object:)
+        # @param object_id [String] the pid for the object
+        def initialize(connection:, version:, object_id:)
```

Comment:
I see this `initialize` code repeated at least 3 times. Can this be extracted into an abstract class?

Reply:
I was reluctant to commit to an inheritance abstraction in order to DRY out `#initialize`. (My "Sandi sense" is tingling.) I'm not sure what the right abstraction is for the client; I'm inclined to pass on this for now if you're OK with that.

Reply:
The repetition feels gross to me, too, for what it's worth. :smiley:

Reply:
I would be more inclined to inline the `Files`, `ReleaseTags`, `SDR`, `Workflow`, and `Workspace` classes within `Object`, tbh.

Reply:
@jcoyne What would you think about :point_up:? I'd like to roll back that abstraction for 1.0.0. Given the above, I question whether that's a helpful abstraction at the moment.